karrot/activities/serializers.py

```diff
@@ -390,10 +436,19 @@ def update(self, activity, validated_data):
                     ParticipantType.objects.create(activity=activity, **entry)
 
         update_data = validated_data.copy()
+
+        if 'is_public' in update_data:
+            if update_data['is_public']:
+                if not activity.public_id:
```

Comment:
```suggestion
        if 'is_public' in update_data and update_data['is_public'] and not activity.public_id:
```

might be nicer with less nesting :)

Reply:
Ah yeah :) An earlier iteration had some `else` clauses... (before I would set `public_id` to `null` when `is_public` was false... but thought it might be nice to keep the id, so you can enable/disable public without the links breaking...).